x-pack/plugins/cloud_integrations/cloud_links/public/maybe_add_cloud_links/theme_darkmode_toggle.test.tsx

```diff
@@ -13,16 +13,20 @@ import { securityMock } from '@kbn/security-plugin/public/mocks';
 
 import { ThemDarkModeToggle } from './theme_darkmode_toggle';
 
-describe('ThemDarkModeToggle', () => {
-  const mockUseUpdateUserProfile = jest.fn();
-  const mockGetSpaceDarkModeValue = jest.fn();
+const mockUseUpdateUserProfile = jest.fn();
+
+jest.mock('@kbn/user-profile-components', () => {
+  const original = jest.requireActual('@kbn/user-profile-components');
+  return {
+    ...original,
+    useUpdateUserProfile: () => mockUseUpdateUserProfile(),
+  };
+});
```

Comment:
NIT: There is a cleaner to do that without mocking the entire module.

```typescript
import * as userProfileComponentsModule from '@kbn/user-profile-components';

// ...

jest.spyOn(userProfileComponentsModule, 'useUpdateUserProfile').mockImplementation(() => {});
```

Reply:
Interesting. I am so used to the `jest.mock` way of doing that I am not sure if I find your alternative "cleaner" 😄 Clearly less verbose but I kind of like having the spy (`mockUseUpdateUserProfile`) out for some reason.